notes.go

```diff
@@ -35,6 +35,7 @@ type NotesService struct {
 // GitLab API docs: https://docs.gitlab.com/ce/api/notes.html
 type Note struct {
 	ID         int    `json:"id"`
+	Type       string `json:"type"`
```

Comment:
the Gitlab documentation does not specify all possible values for this field, so keep a generic `string` type

Reply:
Maybe check the GitLab source code to find out and make a proper type for it? Also I don't see this field in the docs, can you point me to something that shows this field exists?

Reply:
for exemple https://gitlab.com/api/v4/projects/278964/merge_requests/61843/notes, corresponding to this MR https://gitlab.com/gitlab-org/gitlab/-/merge_requests/61843

Reply:
Ok, I did some searching through the GitLab code and think we need to make a type (like this: https://github.com/xanzy/go-gitlab/blob/master/types.go#L27-L41) for these note types:

```
LegacyDiffNote
DiffNote
DiscussionNote
Note
```

Reply:
ok thanks for the pointers. PR updated.